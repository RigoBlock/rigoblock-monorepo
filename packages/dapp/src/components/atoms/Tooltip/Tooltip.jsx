import './Tooltip.scss'
import PropTypes from 'prop-types'
import React from 'react'

export const ICON_SIZES = {
  SMALL: 'small',
  MEDIUM: 'medium',
  LARGE: 'large'
}

const Tooltip = ({ tooltipText }) => {
  return (
    <div className="tooltip-container">
      <div className="tooltip-text">{tooltipText}</div>
      <div className="tooltip-tip" />
    </div>
  )
}

Tooltip.propTypes = {
  tooltipText: PropTypes.string.isRequired
}

export default Tooltip
