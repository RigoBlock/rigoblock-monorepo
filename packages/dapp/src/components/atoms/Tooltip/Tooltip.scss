@import "../../_settings/variables.scss";

.tooltip-container {
  cursor: default;
  user-select: none;
  position: absolute;
  font-family: $font-family;
  font-size: $medium-font-size;
  border-radius: $border-radius;
  color: $black;
  z-index:0;
  bottom: calc(100% + #{$dist-s});
  left: 50%;
  transform: translateX(-50%);
}

.tooltip-text {
  line-height: $line-height;
  border-radius: $border-radius;
  background: $secondary-bg-color;
  padding: $dist-s;
  &:before {
    left: 0;
  }
  &:after {
    right: 0;
  }
}

.tooltip-text:before,
.tooltip-text:after{
  border-radius: $border-radius;
  display: block;
  content: "";
  position: absolute;
  z-index: -100;
  width: 50%;
  top: 0;
  bottom: 0;
  box-shadow: 0 3px 6px rgba(0, 0, 0, 0.1);
}

.tooltip-tip {
  position: absolute;
  z-index: -100;
  width: $dist-s;
  height: $dist-s;
  background: $secondary-bg-color;
  left: 50%;
  bottom: -$dist-xs;
  transform: translateX(-50%) rotate(45deg);
  box-shadow: 3px 3px 6px rgba(0, 0, 0, 0.1);
}
