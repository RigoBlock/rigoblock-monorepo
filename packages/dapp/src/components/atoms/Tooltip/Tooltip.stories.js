import React from 'react'
import { storiesOf } from '@storybook/react'
import '../../_settings/_base.scss'
import Tooltip, { TOOLTIP_SIZES } from './Tooltip'

storiesOf('Atoms/Tooltip', module)
  .add(TOOLTIP_SIZES.SMALL, () => (
    <Tooltip size={TOOLTIP_SIZES.SMALL} type={'help'} to={'/'}>
      Small Tooltip
    </Tooltip>
  ))
  .add(TOOLTIP_SIZES.MEDIUM, () => (
    <Tooltip size={TOOLTIP_SIZES.MEDIUM} type={'help'} to={'/'}>
      Medium Tooltip
    </Tooltip>
  ))
  .add(TOOLTIP_SIZES.LARGE, () => (
    <Tooltip size={TOOLTIP_SIZES.LARGE} type={'help'} to={'/'}>
      Large Tooltip
    </Tooltip>
  ))
