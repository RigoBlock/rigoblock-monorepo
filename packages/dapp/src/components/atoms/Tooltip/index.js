import Tooltip, { TOOLTIP_SIZES } from './Tooltip'

export { TOOLTIP_SIZES }

export default Tooltip
