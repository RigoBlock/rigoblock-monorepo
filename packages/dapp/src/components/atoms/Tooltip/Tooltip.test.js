import { shallow } from 'enzyme'
import Tooltip from './Tooltip.jsx'
import toJson from 'enzyme-to-json'

const props = { type: 'help' }

describe('Tooltip component', () => {
  it('renders correctly', () => {
    expect(
      toJson(shallow(createComponentWithProps(Tooltip, props)))
    ).toMatchSnapshot()
  })
})
