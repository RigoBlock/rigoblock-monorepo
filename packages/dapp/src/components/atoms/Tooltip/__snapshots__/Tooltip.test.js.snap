// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Tooltip component renders correctly 1`] = `
<span
  className="material-icons tooltip medium"
>
  help
</span>
`;
